table/get_context.cc

```diff
@@ -182,7 +182,8 @@ bool GetContext::SaveValue(const ParsedInternalKey& parsed_key,
   assert(matched);
   assert((state_ != kMerge && parsed_key.type != kTypeMerge) ||
          merge_context_ != nullptr);
-  if (ucmp_->Equal(parsed_key.user_key, user_key_)) {
+  if (ucmp_->Equal(parsed_key.user_key, user_key_) ||
+      ucmp_->CompareWithoutTimestamp(parsed_key.user_key, user_key_) == 0) {
```

Comment:
While it is correct to check only key without timestamp for hit, I think merge should only happen if both of user key and timestamp are the same.

Reply:
Yes, merge should also look at timestamp. To RocksDB, two equal keys with different user-timestamps are just two different keys. Without further instruction from the user, e.g. merge policy, RocksDB needs to treat them as separate keys.
I didn't implement merge in this PR though.